### ISSUE TITLE
CODE

### ISSUE THREAD

**Issue by Namitaghadi** [Namitaghadi/jumble-words#1]
import tkinter as tk
from tkinter import filedialog
import pandas as pd 
import numpy as np
from sklearn.naive_bayes import MultinomialNB
from sklearn.svm import LinearSVC
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score, precision_score, recall_score, confusion_matrix
import matplotlib.pyplot as plt

filename = None
AccuracyMeans = []
ErrorrateMeans = []

def browse():
    global filename
    filename = filedialog.askopenfilename()
    print(filename)

def Naive_Bayes():
    if filename:
        global AccuracyMeans, ErrorrateMeans
        df = pd.read_csv(filename) 
        msk = np.random.rand(len(df)) < 0.7
        train = df[msk]
        test = df[~msk]
        testing_data = test.values[:, 0:7]
        testing_data_labels = test.values[:, 8]
        features = train.values[:, 0:7]
        labels = train.values[:, 8].astype('int')
        model = MultinomialNB()
        model.fit(features, labels)
        predictions = model.predict(testing_data)
        
        accuracy = accuracy_score(testing_data_labels, predictions) * 100
        AccuracyMeans.append(accuracy)
        error_rate = 100 - accuracy
        ErrorrateMeans.append(error_rate)
        precision = precision_score(testing_data_labels, predictions) * 100
        recall = recall_score(testing_data_labels, predictions) * 100
        
        print('\n1.Multinomial Naive Bayes :\n')
        print('Confusion Matrix :')
        print(confusion_matrix(testing_data_labels, predictions)) 
        print('Accuracy Is : {:.2f} %'.format(accuracy))
        print('Error Rate Is : {:.2f} %'.format(error_rate))
        print('Precision Is : {:.2f} %'.format(precision))
        print('Recall Is : {:.2f} %\n\n')
        
        labels = ['Error Rate', 'Accuracy']
        sizes = [error_rate, accuracy]
        explode = (0, 0.1)  
        plt.figure()
        plt.pie(sizes, explode=explode, labels=labels, autopct='%1.1f%%', shadow=True, startangle=90)
        plt.title('Naive Bayes Algorithm')
        plt.axis('equal')
        plt.tight_layout()
        plt.show()

def Linear_Svc():
    if filename:
        global AccuracyMeans, ErrorrateMeans
        df = pd.read_csv(filename) 
        msk = np.random.rand(len(df)) < 0.7
        train = df[msk]
        test = df[~msk]
        testing_data = test.values[:, 0:7]
        testing_data_labels = test.values[:, 8]
        features = train.values[:, 0:7]
        labels = train.values[:, 8].astype('int')
        model = LinearSVC()
        model.fit(features, labels)
        predictions = model.predict(testing_data)
        
        accuracy = accuracy_score(testing_data_labels, predictions) * 100
        AccuracyMeans.append(accuracy)
        error_rate = 100 - accuracy
        ErrorrateMeans.append(error_rate)
        precision = precision_score(testing_data_labels, predictions) * 100
        recall = recall_score(testing_data_labels, predictions) * 100
        
        print('\n2.Linear SVC :\n')
        print('Confusion Matrix :')
        print(confusion_matrix(testing_data_labels, predictions)) 
        print('Accuracy Is : {:.2f} %'.format(accuracy))
        print('Error Rate Is : {:.2f} %'.format(error_rate))
        print('Precision Is : {:.2f} %'.format(precision))
        print('Recall Is : {:.2f} %\n\n')

        labels = ['Error Rate', 'Accuracy']
        sizes = [error_rate, accuracy]
        explode = (0, 0.1)  
        plt.figure()
        plt.pie(sizes, explode=explode, labels=labels, autopct='%1.1f%%', shadow=True, startangle=90)
        plt.title('Linear SVC Algorithm')
        plt.axis('equal')
        plt.tight_layout()
        plt.show()
        
def Knn():
    if filename:
        global AccuracyMeans, ErrorrateMeans
        df = pd.read_csv(filename) 
        msk = np.random.rand(len(df)) < 0.7
        train = df[msk]
        test = df[~msk]
        testing_data = test.values[:, 0:7]
        testing_data_labels = test.values[:, 8]
        features = train.values[:, 0:7]
        labels = train.values[:, 8].astype('int')
        model = KNeighborsClassifier(n_neighbors=3)
        model.fit(features, labels)
        predictions = model.predict(testing_data)
        
        accuracy = accuracy_score(testing_data_labels, predictions) * 100
        AccuracyMeans.append(accuracy)
        error_rate = 100 - accuracy
        ErrorrateMeans.append(error_rate)
        precision = precision_score(testing_data_labels, predictions) * 100
        recall = recall_score(testing_data_labels, predictions) * 100
        
        print('\n3.K-Nearest Neighbors  :\n')
        print('Confusion Matrix :')
        print(confusion_matrix(testing_data_labels, predictions)) 
        print('Accuracy Is : {:.2f} %'.format(accuracy))
        print('Error Rate Is : {:.2f} %'.format(error_rate))
        print('Precision Is : {:.2f} %'.format(precision))
        print('Recall Is : {:.2f} %\n\n')

        labels = ['Error Rate', 'Accuracy']
        sizes = [error_rate, accuracy]
        explode = (0, 0.1)  
        plt.figure()
        plt.pie(sizes, explode=explode, labels=labels, autopct='%1.1f%%', shadow=True, startangle=90)
        plt.title('KNN Algorithm')
        plt.axis('equal')
        plt.tight_layout()
        plt.show()

def compare():
    plt.figure()
    plt.bar(['Naive Bayes', 'Linear SVC', 'KNN'], AccuracyMeans, label='Accuracy', color='blue')
    plt.bar(['Naive Bayes', 'Linear SVC', 'KNN'], ErrorrateMeans, label='Error Rate', color='red', bottom=AccuracyMeans)
    plt.xlabel('Classifiers')
    plt.ylabel('Scores')
    plt.title('Performance By Classifiers')
    plt.legend()
    plt.show()

root = tk.Tk()
root.title("Twitter Fake Account Detector")
root.geometry("600x500")
root.resizable(0, 0)

tk.Label(root, 
         text="Fake Account Detector",
         fg="dark violet",
         bg="light blue",
         width=400,
         height=2,
         font="Helvetica 35 bold italic").pack()

browsebutton = tk.Button(root, 
                         text="Browse File", 
                         fg="blue",
                         bg="thistle",
                         width=45,
                         height=2,
                         font="Helvetica 15 bold italic",
                         command=browse)
browsebutton.pack(padx=0, pady=10)

tk.Label(root, 
         text="Classifiers ",
         fg="dark violet",
         bg="light green",
         width=40,
         height=1,
         font="Helvetica 35 bold italic").pack(padx=0, pady=0)

Naive_Bayes_button = tk.Button(root, 
                               text="Naive Bayes", 
                               fg="black",
                               bg="light steel blue",
                               width=25,
                               height=2,
                               font="Helvetica 10 bold italic",
                               command=Naive_Bayes)
Naive_Bayes_button.pack(side=tk.LEFT)

Linear_Svc_Button = tk.Button(root, 
                               text="Linear SVC",
                               fg="black",
                               bg="misty rose",
                               width=25,
                               height=2,
                               font="Helvetica 10 bold italic",
                               command=Linear_Svc)
Linear_Svc_Button.pack(side=tk.LEFT)

KNN_button = tk.Button(root, 
                       text="KNN",
                       fg="black",
                       bg="salmon1",
                       width=25,
                       height=2,
                       font="Helvetica 10 bold italic",
                       command=Knn)
KNN_button.pack(side=tk.LEFT)

All_statistics_Button = tk.Button(root, 
                                  text="Compare",
                                  fg="black",
                                  bg="aquamarine2",
                                  width=35,
                                  height=2,
                                  font="Helvetica 10 bold italic",
                                  command=compare)
All_statistics_Button.place(relx=0.25, rely=0.9)

root.mainloop()
